backpack/__init__.py

```diff
@@ -195,7 +243,17 @@ def extend(module: torch.nn.Module, debug=False):
     module_was_already_extended = getattr(module, "_backpack_extend", False)
     if not module_was_already_extended:
         CTX.add_hook_handle(module.register_forward_hook(hook_store_io))
-        CTX.add_hook_handle(module.register_backward_hook(hook_run_extensions))
+        _register_hooks(module)
         module._backpack_extend = True
 
     return module
+
+
+def _register_hooks(module: Module) -> None:
+    if is_no_op(module):
+        return
```

Comment:
[No action required for now] Discuss where it should be decided that a module will get hooks installed.